### DISCUSSION TITLE
3.1.2 Language of Parts clarification

### DISCUSSION THREAD

**Discussion by EmptyEnd6893** [w3c/wcag#4111] [Q&A]
I have been wondering about the second example for the [Language of Parts understanding page](https://www.w3.org/WAI/WCAG22/Understanding/language-of-parts.html). I have repeated it below for convenience.

'An HTML Web page includes links to versions of the page in other languages (e.g., Deutsch, Français, Nederlands, Catalan, etc.). The text of each link is the name of the language, in that language. The language of each link is indicated via a lang attribute.'

Are the names of the languages not considered proper names in this case? If so, it seems that the criterion would not apply. I realise that language names are slightly unusual in that languages frequently have their own word for other languages, but wouldn't they still count as proper names? Any help would be appreciated.

**Comment by TestPartners** [comment]
No, because they are adjectives, not nouns. For example, England is a proper name, but English isn't.

**Reply by EmptyEnd6893** [reply]
The word 'English' when referring to the language is definitely not an adjective. There are many sources on the Internet that support that the name of languages are proper nouns, such as [Grammarly](https://www.grammarly.com/blog/punctuation-capitalization/capitalization-countries-nationalities-languages/). Most sources I can find use the term proper noun rather than proper name but I think in this case either term can apply. According to the [Wiktionary page for proper name](https://en.wiktionary.org/wiki/proper_name), 'The term is synonymous with proper noun since dictionaries tend to define proper noun as including multi-word phrases. Some uses of proper noun are more restricted.'

**Reply by TestPartners** [reply]
It's an adjective when it's referring to the English (or whatever) version of the website. See https://www.collinsdictionary.com/dictionary/english/english which is rather more authoritative than Grammarly. It can be a noun, such as in the phrase "We speak English", but that's not how it's being used in a language selector.

**Reply by EmptyEnd6893** [reply]
When the word is not presented in a sentence I suppose it is ambiguous whether it is used as an adjective or a noun. I have always understood them to be nouns and the understanding page does say that the 'text of each link is the name of the language, in that language' which I believe supports the interpretation that they are nouns.

**Reply by TestPartners** [reply]
It would help if the Understanding page explained why there is an exception for proper names, but it doesn't and it's not at all obvious to me.

In the case of a language switcher, it seems especially important that screen readers read the language names correctly. It's not such a problem with Latin languages because the pronunciation will just get mangled, but a Latin language TTS synthesizer won't even try to pronounce 繁體中文 so the link would appear to be empty.

**Reply by EmptyEnd6893** [reply]
Perhaps I should create an issue from this Q&A? Some clarifying text on the understanding page may be in order.

**Reply by TestPartners** [reply]
That sounds like a good idea since you are suggesting that Example 2 is wrong because the "lang" attributes would not be required if the names of the languages are proper nouns. In my view, the Understanding page should be changed such that the "lang" attributes are required in Example 2. And the rationale for excluding proper names should be explained if the exclusion is retained, so we can make more informed judgements in ambiguous cases like this.

**Reply by EmptyEnd6893** [reply]
I previously thought that the exception for proper nouns is because most proper nouns remain the same regardless of language, they effectively become a word in all languages (kind of). Similar to the exception for words that have become part of the vernacular of the surrounding text, it is like the word is now part of both languages.

**Reply by EmptyEnd6893** [reply]
Also, I am not saying the example is wrong, only that I don't fully understand it.

**Reply by TestPartners** [reply]
But the pronunciation of most proper nouns doesn't remain the same regardless of language because the phonemes can be entirely different. For instance, when I'm watching subtitled Scandinavian programmes they pronounce Copenhagen as something that sounds like "Choppenen".

**Reply by EmptyEnd6893** [reply]
I know that the pronunciation is not always the same but this fact seems to support the case for not adding a language tag so that the name is pronounced in a way that is familiar for users who do not speak the foreign language. In the third example on the understanding page, there is a good chance that the word 'podcast' is not pronounced the same way in French as it is in English so it would not make sense to me to put an english language tag on the word to throw french screen reader users off. I believe the same principle can be applied to proper nouns that do not change spelling.

**Reply by TestPartners** [reply]
With regard to "podcast", you're talking about a different exception. I agree that that exception makes sense.

But you've got it the wrong way round in the case of the language switcher. It's really important that each language is pronounced correctly for the native speaker of that language, otherwise they may not recognise that it is the link they are looking for. It doesn't matter if the language is pronounced in ways that other people don't understand because they won't be interested in that link.

**Reply by EmptyEnd6893** [reply]
To be clear, I am not talking about nouns that change spelling between languages e.g. Français and French. I am talking about what is probably most proper nouns where the spelling does not change. To give another example, I doubt you would want to add a french language tag every time you use the name 'Paris' in a language that isn't French similarly to how you probably don't want to use english tags when using the word 'podcast' in French. This is why I think the exception for proper nouns exists.

**Reply by TestPartners** [reply]
I understand that, but there are at least three different use cases.

1. A page is mostly in a language that the user speaks and that is the screen reader's default language. The page contains a proper name that is spelled in the user's language. A "lang" attribute is not necessary because the screen reader is likely to pronounce the name much the same way that the user would.
2. A page is mostly in a language that the user speaks and that is the screen reader's default language. The page contains a proper name that is spelled in the foreign language. A "lang" attribute is necessary otherwise the screen reader may mispronounce the name in such as way as to be unrecognisable.
3. A page is mostly in a language that the user does not speak and that the screen reader may or may not support. The page contains a language switcher containing a number of proper names. The names should be spelled in the foreign language. "lang" attributes  are necessary otherwise the screen reader may mispronounce the name in such as way as to be unrecognisable, or the name may not be read at all.

**Reply by EmptyEnd6893** [reply]
I am not disputing any of that, I am only explaining why I think the exception for proper names was made. The apparent issue is that the understanding document does not appear to differentiate between proper names that are written differently in different languages and names that are written the same, at least as far as I can tell.
